tests/CatalogTests/Helpers/CatalogPropertiesTests.cs

```diff
@@ -31,5 +36,196 @@ public void Constructor_InitializesPropertiesWithNonNullValues()
+        {
+            var exception = await Assert.ThrowsAsync<ArgumentNullException>(
+                () => CatalogProperties.ReadAsync(
+                    storage: Mock.Of<IStorage>(),
+                    telemetryService: null,
+                    cancellationToken: CancellationToken.None));
+
+            Assert.Equal("telemetryService", exception.ParamName);
+        }
+
+        [Fact]
+        public async Task ReadAsync_ThrowsIfCancelled()
+        {
+            await Assert.ThrowsAsync<OperationCanceledException>(
+                () => CatalogProperties.ReadAsync(
+                    Mock.Of<IStorage>(),
+                    Mock.Of<ITelemetryService>(),
+                    new CancellationToken(canceled: true)));
+        }
+
+        [Fact]
+        public async Task ReadAsync_ReturnsNullPropertiesIfStorageReturnsNull()
+        {
+            var storage = CreateStorageMock(json: null);
+
+            var catalogProperties = await CatalogProperties.ReadAsync(
+                storage.Object,
+                Mock.Of<ITelemetryService>(),
+                CancellationToken.None);
+
+            Assert.NotNull(catalogProperties);
+            Assert.Null(catalogProperties.LastCreated);
+            Assert.Null(catalogProperties.LastDeleted);
+            Assert.Null(catalogProperties.LastEdited);
+
+            storage.Verify();
+        }
+
+        [Fact]
+        public async Task ReadAsync_ReturnsNullPropertiesIfPropertiesMissing()
+        {
+            var storage = CreateStorageMock(json: "{}");
+
+            var catalogProperties = await CatalogProperties.ReadAsync(
+                storage.Object,
+                Mock.Of<ITelemetryService>(),
+                CancellationToken.None);
+
+            Assert.NotNull(catalogProperties);
+            Assert.Null(catalogProperties.LastCreated);
+            Assert.Null(catalogProperties.LastDeleted);
+            Assert.Null(catalogProperties.LastEdited);
+
+            storage.Verify();
+        }
+
+        [Fact]
+        public async Task ReadAsync_ReturnsAllPropertiesIfAllPropertiesSet()
+        {
+            var datetimeOffset = CreateDateTimeOffset(TimeSpan.Zero);
+            var json = $"{{\"nuget:lastCreated\":\"{(datetimeOffset.ToString("O"))}\"," +
+                $"\"nuget:lastDeleted\":\"{(datetimeOffset.ToString("O"))}\"," +
+                $"\"nuget:lastEdited\":\"{(datetimeOffset.ToString("O"))}\"}}";
+            var storage = CreateStorageMock(json);
+
+            var catalogProperties = await CatalogProperties.ReadAsync(
+                storage.Object,
+                Mock.Of<ITelemetryService>(),
+                CancellationToken.None);
+
+            Assert.NotNull(catalogProperties);
+            Assert.NotNull(catalogProperties.LastCreated);
+            Assert.NotNull(catalogProperties.LastDeleted);
+            Assert.NotNull(catalogProperties.LastEdited);
+            Assert.Equal(datetimeOffset, catalogProperties.LastCreated.Value);
```

Comment:
different timestamp values to be used to verify that they aren't getting mixed up

Reply:
Well, considering these are moved, feel free to ignore. But now's a good as time as any IMO.

Reply:
Assuming this is what you meant, I think I handled it as part of https://github.com/NuGet/NuGet.Services.Metadata/pull/414/commits/c9eea547a693d526e0bea2053a9240a503d5b10e